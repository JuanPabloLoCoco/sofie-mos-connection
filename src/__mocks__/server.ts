import { EventEmitter } from 'events'

import { Server } from 'net'
import { SocketMock } from './socket'
// import { Writable } from 'stream'

const iconv = require('iconv-lite')
iconv.encodingExists('utf16-be')

// Mock the Server class in 'net':
let setTimeoutOrg = setTimeout

let instances: Array<IServerMock> = []

export class ServerMock extends EventEmitter implements Server {
	// Mock implementation:

	maxConnections: number
	connections: number
	listening: boolean
	listenToPort: number

	private _responses: Array<(data: any) => string | Buffer > = []

	constructor () {
		super()

		// @ts-ignore this is comparable with ISocketMock
		instances.push(this)

		this.listen = jest.fn(this.listen)
		this.close = jest.fn(this.close)
		this.address = jest.fn(this.address)
		this.getConnections = jest.fn(this.getConnections)
		this.ref = jest.fn(this.ref)
		this.unref = jest.fn(this.unref)
		// this.addListener = jest.fn(this.addListener)
		// this.emit = jest.fn(this.emit)
		// this.on = jest.fn(this.on)
		// this.once = jest.fn(this.once)
		// this.prependListener = jest.fn(this.prependListener)
		// this.prependOnceListener = jest.fn(this.prependOnceListener)

	}
	static mockClear () {
		instances.splice(0, 9999)
	}
	static get instances () {
		return instances
	}

	listen (port: any) {

		this.listenToPort = port
		setTimeoutOrg(() => {
			this.listening = true
			this.emit('listening', {})
		},1)
		return this
	}
	close (callback?: Function) {
		if (callback) callback()
		setTimeoutOrg(() => {
			this.emit('close')
		},1)
		return this
	}
	address () { return { port: 0, family: 'string', address: 'string' } }
	getConnections (cb: (error: Error | null, count: number) => void) { cb(null, 0) }
	ref () { return this }
	unref () { return this }
	// addListener (event: string, listener: (...args: any[]) => void) { return this }
	// emit (event: string | symbol, ...args: any[]) { return true }
	// on (event: string, listener: (...args: any[]) => void) { return this }
	// once (event: string, listener: (...args: any[]) => void) { return this }
	// prependListener (event: string, listener: (...args: any[]) => void) { return this }
	// prependOnceListener (event: string, listener: (...args: any[]) => void) { return this }
	// ------------------------------------------------------------------------
	// Mock methods:
	mockNewConnection () {
		// "Someone has connected"
		let socket = new SocketMock()
		socket.on('data', (d) => {
			this.emit('data', d)
		})

		this.emit('connection', socket)

		return socket
	}

	mockSentMessage (data: any, encoding: any) {
		encoding = encoding
		if (this._responses.length) {
			// send reply:

			let cb: any = this._responses.shift()
			let msg

			setTimeoutOrg(() => {

				if (typeof cb === 'string') {
					msg = cb
				} else {
					msg = cb(data)
				}

				this.mockReceiveMessage(msg)
			},1)
		}
	}
	mockReceiveMessage (msg: string | Buffer) {
		this.emit('data', msg)
	}
	mockAddReply (cb: (data: any) => string | Buffer) {
		this._responses.push(cb)
	}
	mockClear () {
		this._responses.splice(0, 9999)
	}
	decode (data: Buffer): string {
		return iconv.decode(data, 'utf16-be')
	}
	encode (str: string) {
		return iconv.encode(str, 'utf16-be')
	}
}

export type IServerMock = jest.Mocked<ServerMock>
