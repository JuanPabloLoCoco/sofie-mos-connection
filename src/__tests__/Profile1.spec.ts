import {
	checkMessageSnapshot,
	clearMocks,
	decode,
	delay,
	doBeforeAll,
	encode,
	fakeIncomingMessage,
	fixSnapshot,
	getMosConnection,
	getMosDevice,
	getXMLReply,
	setupMocks,
} from './lib'
import { MosConnection, MosDevice, IMOSObject, IMOSListMachInfo } from '..'
import { SocketMock } from '../__mocks__/socket'
import { xmlData, xmlApiData } from '../__mocks__/testData'
import { xml2js } from 'xml-js'
import * as Utils from '../utils/Utils'

// @ts-ignore imports are unused
import { Socket } from 'net'

beforeAll(() => {
	setupMocks()
})
beforeEach(() => {
	clearMocks()
})
describe('Profile 1', () => {
	let mosDevice: MosDevice
	let mosConnection: MosConnection

	let socketMockLower: SocketMock
	let socketMockUpper: SocketMock
	let socketMockQuery: SocketMock

	let serverSocketMockLower: SocketMock
	let serverSocketMockUpper: SocketMock
	let serverSocketMockQuery: SocketMock

	let onRequestMachineInfo: jest.Mock<any, any>
	let onRequestMOSObject: jest.Mock<any, any>
	let onRequestAllMOSObjects: jest.Mock<any, any>

	beforeAll(async () => {
		mosConnection = await getMosConnection(
			{
				'0': true,
				'1': true,
			},
			true
		)
		mosDevice = await getMosDevice(mosConnection)

		// Profile 0:
		onRequestMachineInfo = jest.fn(() => {
			return Promise.resolve(xmlApiData.machineInfo)
		})
		mosDevice.onRequestMachineInfo((): Promise<IMOSListMachInfo> => {
			return onRequestMachineInfo()
		})
		// Profile 1:
		onRequestMOSObject = jest.fn(() => {
			return Promise.resolve(xmlApiData.mosObj)
		})
		onRequestAllMOSObjects = jest.fn(() => {
			return Promise.resolve([xmlApiData.mosObj, xmlApiData.mosObj2])
		})
		mosDevice.onRequestMOSObject((objId: string): Promise<IMOSObject | null> => {
			return onRequestMOSObject(objId)
		})
		mosDevice.onRequestAllMOSObjects((): Promise<Array<IMOSObject>> => {
			return onRequestAllMOSObjects()
		})
		let b = doBeforeAll()
		socketMockLower = b.socketMockLower
		socketMockUpper = b.socketMockUpper
		socketMockQuery = b.socketMockQuery
		serverSocketMockLower = b.serverSocketMockLower
		serverSocketMockUpper = b.serverSocketMockUpper
		serverSocketMockQuery = b.serverSocketMockQuery
	})
	afterAll(async () => {
		await mosDevice.dispose()
		await mosConnection.dispose()
	})
	beforeEach(() => {
		// SocketMock.mockClear()
		onRequestMOSObject.mockClear()
		onRequestAllMOSObjects.mockClear()

		serverSocketMockLower.mockClear()
		serverSocketMockUpper.mockClear()
		if (serverSocketMockQuery) serverSocketMockQuery.mockClear()
		socketMockLower.mockClear()
		socketMockUpper.mockClear()
		if (socketMockQuery) socketMockQuery.mockClear()
	})
	test('init', async () => {
		expect(mosDevice).toBeTruthy()
		expect(socketMockLower).toBeTruthy()
		expect(socketMockUpper).toBeTruthy()
		mosDevice.checkProfileValidness()
	})

	test('onRequestMOSObject', async () => {
		// Fake incoming message on socket:
		await fakeIncomingMessage(serverSocketMockLower, xmlData.reqObj)
		expect(onRequestMOSObject).toHaveBeenCalledTimes(1)
		expect(onRequestMOSObject.mock.calls[0][0]).toEqual(xmlApiData.mosObj.ID!.toString())
		expect(fixSnapshot(onRequestMOSObject.mock.calls)).toMatchSnapshot()
		expect(onRequestAllMOSObjects).toHaveBeenCalledTimes(0)

		// Check reply to socket server:
		await serverSocketMockLower.mockWaitForSentMessages()
		expect(serverSocketMockLower.mockSentMessage).toHaveBeenCalledTimes(1)
		// @ts-ignore mock
		let reply = decode(serverSocketMockLower.mockSentMessage['mock'].calls![0][0])
		let parsedReply: any = xml2js(reply, { compact: true, nativeType: true, trim: true })

		expect(parsedReply.mos.mosObj.objID._text + '').toEqual(xmlApiData.mosObj.ID!.toString())
		expect(parsedReply.mos.mosObj.objSlug._text + '').toEqual(xmlApiData.mosObj.Slug.toString())
		expect(parsedReply).toMatchSnapshot()
	})
	test('onRequestAllMOSObjects', async () => {
		let mockReply = jest.fn((data) => {
			let str = decode(data)
			let messageID = str.match(/<messageID>([^<]+)<\/messageID>/)![1]
			let repl = getXMLReply(messageID, '<mosAck></mosAck>')

			return encode(repl)
		})
		socketMockLower.mockAddReply(mockReply)

		// Fake incoming message on socket:
		await fakeIncomingMessage(serverSocketMockLower, xmlData.mosReqAll)
		expect(onRequestAllMOSObjects).toHaveBeenCalledTimes(1)
		expect(fixSnapshot(onRequestAllMOSObjects.mock.calls)).toMatchSnapshot()
		expect(onRequestMOSObject).toHaveBeenCalledTimes(0)

		// Check reply to socket server:
		await serverSocketMockLower.mockWaitForSentMessages()
		expect(serverSocketMockLower.mockSentMessage).toHaveBeenCalledTimes(1)
		// @ts-ignore mock
		let reply = decode(serverSocketMockLower.mockSentMessage.mock.calls[0][0])
		let parsedReply: any = xml2js(reply, { compact: true, nativeType: true, trim: true })
		expect(parsedReply.mos.mosAck).toBeTruthy()
		expect(parsedReply).toMatchSnapshot()

		// @ts-ignore mock
		serverSocketMockLower.mockSentMessage.mockClear()
		await serverSocketMockLower.mockWaitForSentMessages()
		await delay(100) // to allow for async timers & events to triggered

		expect(mockReply).toHaveBeenCalledTimes(1)

		const sentData = Utils.xml2js(decode(mockReply.mock.calls[0][0])) as any

		expect(sentData.mos.mosListAll.mosObj).toHaveLength(2)
		expect(sentData.mos.mosListAll.mosObj[0].objID + '').toEqual(xmlApiData.mosObj.ID!.toString())
		expect(sentData.mos.mosListAll.mosObj[0].objSlug + '').toEqual(xmlApiData.mosObj.Slug.toString())
		expect(sentData.mos.mosListAll.mosObj[1].objID + '').toEqual(xmlApiData.mosObj2.ID!.toString())
		expect(sentData.mos.mosListAll.mosObj[1].objSlug + '').toEqual(xmlApiData.mosObj2.Slug.toString())
		sentData.mos.messageID = 99999 // not important
		expect(sentData).toMatchSnapshot()
	})
	test('getMOSObject', async () => {
		// Prepare mock server response:
		let mockReply = jest.fn((data) => {
			let str = decode(data)
			let messageID = str.match(/<messageID>([^<]+)<\/messageID>/)![1]
			let repl = getXMLReply(messageID, xmlData.mosObj)

			return encode(repl)
		})
		socketMockLower.mockAddReply(mockReply)
		let returnedObj: IMOSObject = await mosDevice.sendRequestMOSObject(xmlApiData.mosObj.ID!)
		expect(mockReply).toHaveBeenCalledTimes(1)
		let msg = decode(mockReply.mock.calls[0][0])
		expect(msg).toMatch(/<mosReqObj>/)
		checkMessageSnapshot(msg)

		expect(returnedObj).toMatchObject(xmlApiData.roList)
		expect(returnedObj).toMatchSnapshot()
	})
	test('getAllMOSObjects', async () => {
		expect(socketMockLower).toBeTruthy()
		// Prepare mock server response:
		let mockReply = jest.fn((data) => {
			let str = decode(data)
			let messageID = str.match(/<messageID>([^<]+)<\/messageID>/)![1]
			let repl = getXMLReply(messageID, xmlData.mosListAll)
			return encode(repl)
		})
		socketMockLower.mockAddReply(mockReply)
		let returnedObjs: Array<IMOSObject> = await mosDevice.sendRequestAllMOSObjects()

		expect(mockReply).toHaveBeenCalledTimes(1)
		let msg = decode(mockReply.mock.calls[0][0])
		expect(msg).toMatch(/<mosReqAll>/)
		checkMessageSnapshot(msg)

		expect(returnedObjs).toMatchObject(xmlApiData.mosListAll)
		expect(returnedObjs).toMatchSnapshot()
	})
})
