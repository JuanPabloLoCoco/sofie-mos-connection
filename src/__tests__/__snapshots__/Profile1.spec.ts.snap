// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile 1 getAllMOSObjects 1`] = `
"<mos>
  <ncsID>their.mos.id</ncsID>
  <mosID>our.mos.id</mosID>
  <messageID>xx</messageID>
  <mosReqAll>
    <pause>0</pause>
  </mosReqAll>
</mos>"
`;

exports[`Profile 1 getAllMOSObjects 2`] = `
Array [
  Object {
    "AirStatus": undefined,
    "Changed": MosTime {
      "_time": 2009-11-01T14:35:55.000Z,
      "_timezone": "Z",
      "_timezoneDeclaration": "",
      "_timezoneZuluIndicator": "Z",
    },
    "ChangedBy": MosString128 {
      "_str": "Chris",
      "strict": true,
    },
    "Created": MosTime {
      "_time": 2009-10-31T23:39:12.000Z,
      "_timezone": "Z",
      "_timezoneDeclaration": "",
      "_timezoneZuluIndicator": "Z",
    },
    "CreatedBy": MosString128 {
      "_str": "Chris",
      "strict": true,
    },
    "Description": Object {
      "elements": Array [
        Object {
          "$name": "p",
          "$type": "element",
          "elements": Array [
            Object {
              "$type": "text",
              "text": "Exterior footage of",
            },
            Object {
              "$name": "em",
              "$type": "text",
              "text": "Baley Park Hotel",
            },
            Object {
              "$type": "text",
              "text": "on fire with natural sound. Trucks are visible for the first portion of the clip.",
            },
            Object {
              "$name": "em",
              "$type": "text",
              "text": "CG locator at 0:04 and duration 0:05, Baley Park Hotel.",
            },
          ],
        },
        Object {
          "$name": "p",
          "$type": "element",
          "elements": Array [
            Object {
              "$name": "tab",
              "$type": "element",
            },
            Object {
              "$type": "text",
              "text": "Cuts to view of fire personnel exiting hotel lobby and cleaning up after the fire is out.",
            },
          ],
        },
        Object {
          "$name": "p",
          "$type": "text",
          "em": "Clip has been doubled for pad on voice over.",
          "text": "Clip has been doubled for pad on voice over.",
        },
      ],
    },
    "Duration": undefined,
    "Group": undefined,
    "ID": MosString128 {
      "_str": "M000123",
      "strict": true,
    },
    "MosAbstract": undefined,
    "Paths": Array [
      Object {
        "Description": "MPEG2 Video",
        "Target": "\\\\server\\\\media\\\\clip392028cd2320s0d.mxf",
        "Type": "PATH",
      },
      Object {
        "Description": "WM9 750Kbps",
        "Target": "http://server/proxy/clipe.wmv",
        "Type": "PROXY PATH",
      },
      Object {
        "Description": "MOS Object",
        "Target": "http://server/proxy/clipe.xml",
        "Type": "METADATA PATH",
      },
    ],
    "Revision": undefined,
    "Slug": MosString128 {
      "_str": "HOTEL FIRE",
      "strict": true,
    },
    "Status": undefined,
    "TimeBase": undefined,
    "Type": undefined,
  },
  Object {
    "AirStatus": "READY",
    "Changed": MosTime {
      "_time": 2009-11-01T15:21:15.000Z,
      "_timezone": "Z",
      "_timezoneDeclaration": "",
      "_timezoneZuluIndicator": "Z",
    },
    "ChangedBy": MosString128 {
      "_str": "Chris",
      "strict": true,
    },
    "Created": MosTime {
      "_time": 2009-11-01T15:19:01.000Z,
      "_timezone": "Z",
      "_timezoneDeclaration": "",
      "_timezoneZuluIndicator": "Z",
    },
    "CreatedBy": MosString128 {
      "_str": "Phil",
      "strict": true,
    },
    "Description": "VOICE OVER MATERIAL OF COLSTAT MURDER SITES SHOT ON 1-NOV.",
    "Duration": 800,
    "Group": undefined,
    "ID": MosString128 {
      "_str": "M000224",
      "strict": true,
    },
    "MosAbstract": undefined,
    "MosExternalMetaData": Array [
      Object {
        "MosPayload": Object {
          "Approved": 0,
          "Creator": "SHOLMES",
          "ModBy": "LJOHNSTON",
          "ModTime": 20010308142001,
          "Owner": "SHOLMES",
          "TextTime": 278,
          "mediaTime": 0,
          "text": 0,
        },
        "MosSchema": "http://MOSA4.com/mos/supported_schemas/MOSAXML2.08",
        "MosScope": "STORY",
      },
    ],
    "Paths": Array [
      Object {
        "Description": "MPEG2 Video",
        "Target": "\\\\server\\\\media\\\\clip392028cd2320s0d.mxf",
        "Type": "PATH",
      },
      Object {
        "Description": "WM9 750Kbps",
        "Target": "http://server/proxy/clipe.wmv",
        "Type": "PROXY PATH",
      },
      Object {
        "Description": "MOS Object",
        "Target": "http://server/proxy/clipe.xml",
        "Type": "METADATA PATH",
      },
    ],
    "Revision": 4,
    "Slug": MosString128 {
      "_str": "COLSTAT MURDER:VO",
      "strict": true,
    },
    "Status": "UPDATED",
    "TimeBase": 59.94,
    "Type": "VIDEO",
  },
]
`;

exports[`Profile 1 getMOSObject 1`] = `
"<mos>
  <ncsID>their.mos.id</ncsID>
  <mosID>our.mos.id</mosID>
  <messageID>xx</messageID>
  <mosReqObj>
    <objID>M000123</objID>
  </mosReqObj>
</mos>"
`;

exports[`Profile 1 getMOSObject 2`] = `
Object {
  "AirStatus": "READY",
  "Changed": MosTime {
    "_time": 2009-10-31T23:39:12.000Z,
    "_timezone": "Z",
    "_timezoneDeclaration": "",
    "_timezoneZuluIndicator": "Z",
  },
  "ChangedBy": MosString128 {
    "_str": "Chris",
    "strict": true,
  },
  "Created": MosTime {
    "_time": 2009-10-31T23:39:12.000Z,
    "_timezone": "Z",
    "_timezoneDeclaration": "",
    "_timezoneZuluIndicator": "Z",
  },
  "CreatedBy": MosString128 {
    "_str": "Chris",
    "strict": true,
  },
  "Description": Object {
    "elements": Array [
      Object {
        "$name": "p",
        "$type": "element",
        "elements": Array [
          Object {
            "$type": "text",
            "text": "Exterior footage of",
          },
          Object {
            "$name": "em",
            "$type": "text",
            "text": "Baley Park Hotel",
          },
          Object {
            "$type": "text",
            "text": "on fire with natural sound. Trucks are visible for the first portion of the clip.",
          },
          Object {
            "$name": "em",
            "$type": "text",
            "text": "CG locator at 0:04 and duration 0:05, Baley Park Hotel.",
          },
        ],
      },
      Object {
        "$name": "p",
        "$type": "element",
        "elements": Array [
          Object {
            "$name": "tab",
            "$type": "element",
          },
          Object {
            "$type": "text",
            "text": "Cuts to view of fire personnel exiting hotel lobby and cleaning up after the fire is out.",
          },
        ],
      },
      Object {
        "$name": "p",
        "$type": "text",
        "em": "Clip has been doubled for pad on voice over.",
        "text": "Clip has been doubled for pad on voice over.",
      },
    ],
  },
  "Duration": 1800,
  "Group": "Show 7",
  "ID": MosString128 {
    "_str": "M000123",
    "strict": true,
  },
  "MosAbstract": Object {
    "elements": Array [
      Object {
        "$name": "b",
        "$type": "text",
        "text": "Hotel Fire",
      },
      Object {
        "$name": "em",
        "$type": "text",
        "text": "vo",
      },
      Object {
        "$type": "text",
        "text": ":30",
      },
    ],
  },
  "MosExternalMetaData": Array [
    Object {
      "MosPayload": Object {
        "Approved": 0,
        "Creator": "SHOLMES",
        "ModBy": "LJOHNSTON",
        "ModTime": 20010308142001,
        "Owner": "SHOLMES",
        "TextTime": 278,
        "mediaTime": 0,
        "text": 0,
      },
      "MosSchema": "http://MOSA4.com/mos/supported_schemas/MOSAXML2.08",
      "MosScope": "STORY",
    },
  ],
  "Paths": Array [
    Object {
      "Description": "MPEG2 Video",
      "Target": "\\\\server\\\\media\\\\clip392028cd2320s0d.mxf",
      "Type": "PATH",
    },
    Object {
      "Description": "WM9 750Kbps",
      "Target": "http://server/proxy/clipe.wmv",
      "Type": "PROXY PATH",
    },
    Object {
      "Description": "MOS Object",
      "Target": "http://server/proxy/clipe.xml",
      "Type": "METADATA PATH",
    },
  ],
  "Revision": 1,
  "Slug": MosString128 {
    "_str": "Hotel Fire",
    "strict": true,
  },
  "Status": "NEW",
  "TimeBase": 59.94,
  "Type": "VIDEO",
}
`;

exports[`Profile 1 onRequestAllMOSObjects 1`] = `
Array [
  Array [],
]
`;

exports[`Profile 1 onRequestAllMOSObjects 2`] = `
Object {
  "mos": Object {
    "messageID": Object {
      "_text": 1634,
    },
    "mosAck": Object {
      "objID": Object {
        "_text": 0,
      },
      "objRev": Object {
        "_text": 0,
      },
      "status": Object {
        "_text": "ACK",
      },
      "statusDescription": Object {},
    },
    "mosID": Object {
      "_text": "our.mos.id",
    },
    "ncsID": Object {
      "_text": "their.mos.id",
    },
  },
}
`;

exports[`Profile 1 onRequestAllMOSObjects 3`] = `
Object {
  "mos": Object {
    "messageID": 99999,
    "mosID": "our.mos.id",
    "mosListAll": Object {
      "mosObj": Array [
        Object {
          "created": "2001-01-01T00:00:00,000",
          "createdBy": "Jonas",
          "mosExternalMetadata": Object {
            "mosPayload": Object {
              "Approved": 0,
              "Creator": "SHOLMES",
              "ModBy": "LJOHNSTON",
              "ModTime": 20010308142001,
              "Owner": "SHOLMES",
              "TextTime": 278,
              "mediaTime": 0,
              "text": 0,
            },
            "mosSchema": "http://MOSA4.com/mos/supported_schemas/MOSAXML2.08</mosSchema>",
            "mosScope": "STORY",
          },
          "objAir": "READY",
          "objDur": 1500,
          "objID": "M000123",
          "objPaths": Object {
            "objMetadataPath": Object {
              "techDescription": "MOS Object",
              "text": "http://server/proxy/clipe.xml",
            },
            "objPath": Object {
              "techDescription": "MPEG2 Video",
              "text": "\\\\server\\\\media\\\\clip392028cd2320s0d.mxf",
            },
            "objProxyPath": Object {
              "techDescription": "WM9 750Kbps",
              "text": "http://server/proxy/clipe.wmv",
            },
          },
          "objRev": 1,
          "objSlug": "My new object",
          "objTB": 50,
          "objType": "VIDEO",
          "status": "READY",
        },
        Object {
          "created": "2001-01-01T00:00:00,000",
          "createdBy": "Jonas",
          "objAir": "READY",
          "objDur": 1000,
          "objID": "M0003523",
          "objPaths": Object {
            "objMetadataPath": Object {
              "techDescription": "MOS Object",
              "text": "http://server/proxy/clipe.xml",
            },
            "objPath": Object {
              "techDescription": "MPEG2 Video",
              "text": "\\\\server\\\\media\\\\clip392028cd2320s0d.mxf",
            },
            "objProxyPath": Object {
              "techDescription": "WM9 750Kbps",
              "text": "http://server/proxy/clipe.wmv",
            },
          },
          "objRev": 1,
          "objSlug": "My new object 2",
          "objTB": 50,
          "objType": "VIDEO",
          "status": "READY",
        },
      ],
    },
    "ncsID": "their.mos.id",
  },
}
`;

exports[`Profile 1 onRequestMOSObject 1`] = `
Array [
  Array [
    "M000123",
  ],
]
`;

exports[`Profile 1 onRequestMOSObject 2`] = `
Object {
  "mos": Object {
    "messageID": Object {
      "_text": 1633,
    },
    "mosID": Object {
      "_text": "our.mos.id",
    },
    "mosObj": Object {
      "created": Object {
        "_text": "2001-01-01T00:00:00,000",
      },
      "createdBy": Object {
        "_text": "Jonas",
      },
      "mosExternalMetadata": Object {
        "mosPayload": Object {
          "Approved": Object {
            "_text": 0,
          },
          "Creator": Object {
            "_text": "SHOLMES",
          },
          "ModBy": Object {
            "_text": "LJOHNSTON",
          },
          "ModTime": Object {
            "_text": 20010308142001,
          },
          "Owner": Object {
            "_text": "SHOLMES",
          },
          "TextTime": Object {
            "_text": 278,
          },
          "mediaTime": Object {
            "_text": 0,
          },
        },
        "mosSchema": Object {
          "_text": "http://MOSA4.com/mos/supported_schemas/MOSAXML2.08</mosSchema>",
        },
        "mosScope": Object {
          "_text": "STORY",
        },
      },
      "objAir": Object {
        "_text": "READY",
      },
      "objDur": Object {
        "_text": 1500,
      },
      "objID": Object {
        "_text": "M000123",
      },
      "objPaths": Object {
        "objMetadataPath": Object {
          "_attributes": Object {
            "techDescription": "MOS Object",
          },
          "_text": "http://server/proxy/clipe.xml",
        },
        "objPath": Object {
          "_attributes": Object {
            "techDescription": "MPEG2 Video",
          },
          "_text": "\\\\server\\\\media\\\\clip392028cd2320s0d.mxf",
        },
        "objProxyPath": Object {
          "_attributes": Object {
            "techDescription": "WM9 750Kbps",
          },
          "_text": "http://server/proxy/clipe.wmv",
        },
      },
      "objRev": Object {
        "_text": 1,
      },
      "objSlug": Object {
        "_text": "My new object",
      },
      "objTB": Object {
        "_text": 50,
      },
      "objType": Object {
        "_text": "VIDEO",
      },
      "status": Object {
        "_text": "READY",
      },
    },
    "ncsID": Object {
      "_text": "their.mos.id",
    },
  },
}
`;

exports[`Profile 1 sendMOSObject 1`] = `
"<mos>
  <ncsID>their.mos.id</ncsID>
  <mosID>our.mos.id</mosID>
  <messageID>xx</messageID>
  <mosObj>
    <objID>M000123</objID>
    <objSlug>theslug</objSlug>
    <mosAbstract>abstract!</mosAbstract>
    <objGroup>grp</objGroup>
    <objType>VIDEO</objType>
    <objTB>50</objTB>
    <objRev>5</objRev>
    <objDur>1000</objDur>
    <status>NOT READY</status>
    <objAir>READY</objAir>
    <objPaths>
      <objPath techDescription=\\"path\\">folder/target</objPath>
    </objPaths>
    <createdBy>me!</createdBy>
    <created>2005-07-01T15:23:18,000Z</created>
  </mosObj>
</mos>"
`;
